packages/lib/src/interactions/SelectionTool.tsx

```diff
@@ -27,16 +28,31 @@ function SelectionTool(props: Props) {
     id = 'Selection',
     modifierKey,
     disabled,
+    clampCenter,
   } = props;
 
   const camera = useThree((state) => state.camera);
   const { worldToData } = useAxisSystemContext();
   const shouldInteract = useRegisterInteraction(id, { modifierKey, disabled });
 
-  const [startPoint, setStartPoint] = useState<Vector2>();
-  const [endPoint, setEndPoint] = useRafState<Vector2 | undefined>(undefined);
+  const [originPoint, setOriginPoint] = useState<Vector2>();
+  const [selection, setSelection] = useRafState<Selection | undefined>(
+    undefined
+  );
```

Comment:
As `startPoint` and `endPoint` are updated together, it is more handy to work with a single state. It also simplifies the syntax at several places below.